runtime/hydradx/src/evm/evm_fee.rs

```diff
@@ -87,10 +102,52 @@ where
 			return Ok(None);
 		}
 		let account_id = T::AddressMapping::into_account_id(*who);
-		let fee_currency = AC::get(&account_id);
-		let Some((converted, price)) = C::convert((EC::get(), fee_currency, fee.unique_saturated_into())) else {
-			return Err(Error::<T>::WithdrawFailed);
-		};
+		let account_fee_currency = AccountCurrency::get(&account_id);
+
+		let dot = DotAssetId::get();
+
+		let (converted, fee_currency, price) =
+			if SwappablePaymentAssetSupport::is_transaction_fee_currency(account_fee_currency) {
+				let Some((converted, price)) =
+					C::convert((EvmFeeAsset::get(), account_fee_currency, fee.unique_saturated_into()))
+				else {
+					return Err(Error::<T>::WithdrawFailed);
+				};
+				(converted, account_fee_currency, price)
+			} else {
+				//In case of insufficient asset we buy DOT with insufficient asset, and using that DOT and amount as fee currency
+				let Some((_, eth_dot_price)) = C::convert((EvmFeeAsset::get(), dot, fee.unique_saturated_into()))
+				else {
+					return Err(Error::<T>::WithdrawFailed);
+				};
+
+				let Some(eth_dot_price_as_fixed) = FixedU128::checked_from_rational(eth_dot_price.n, eth_dot_price.d)
+				else {
+					return Err(Error::<T>::WithdrawFailed);
+				};
+
+				let fee_in_dot: Balance = convert_fee_with_price(fee.unique_saturated_into(), eth_dot_price_as_fixed)
```

Comment:
why you need to do this ? 

isnt this result from the convert function on line 119 ?

Reply:
good catch, conver already returns this.